control/autopilot/include/autopilot/autopilot_inl.h

```diff
@@ -1107,13 +1170,20 @@ quadrotor_common::ControlCommand AutoPilot::followReference(
 
   reference_state_ = quadrotor_common::TrajectoryPoint(reference_state_input_);
 
+
+  reference_trajectory_ = quadrotor_common::Trajectory();
+  reference_trajectory_.trajectory_type = 
+    quadrotor_common::Trajectory::TrajectoryType::GENERAL;
+  reference_trajectory_.points.push_back(reference_state_);
   const quadrotor_common::ControlCommand command = base_controller_.run(
-      state_estimate, reference_state_, base_controller_params_);
+      state_estimate, reference_trajectory_, base_controller_params_);
 
   return command;
 }
 
-quadrotor_common::ControlCommand AutoPilot::executeTrajectory(
+template <typename Tcontroller, typename Tparams>
+quadrotor_common::ControlCommand
+AutoPilot<Tcontroller, Tparams>::executeTrajectory(
     const quadrotor_common::QuadStateEstimate& state_estimate,
     ros::Duration* trajectory_execution_left_duration,
```

Comment:
general comment: really a point to a ros::Duration and int ?!? ---> how about a smart pointer?

Reply:
meh... lets make it pass-by-reference!
then there's no pointer magic but still avoiding copying, ok?

Reply:
No, it is our convention that arguments which are changed by the function are passed as pointers.

Reply:
Conventions are here to be respected. Ok

Reply:
agree, mark as done